controllers/cart.js

```diff
@@ -31,4 +31,65 @@ module.exports = (app) => {
       }
     });
   });
+
+  app.get('/cart/grocery-list', (req, res) => {
+    // TODO: (1) Find user's favorites (2) show all of them
+    if (app.locals.user) {
+      userId = app.locals.user.id;
+      UserSchema.findById(userId, function(err, user) {
+        if (err) { console.error(err) }
+        // to get updated user object
+        RecipeSchema.find()
+          .where('_id')
+          .in(user.recipesInCart)
+          .then(function(cartRecipes) {
+            let ingredients = getIngredients(cartRecipes);
+            res.render('grocery-list', {
+              ingredients: ingredients
+            });
+          })
+          .catch(function(err) {
+            console.log('error: ' + err);
+          })
+      });
+    } else {
+      res.render('grocery-list');
+    }
+  });
+
+  function getIngredients(cartRecipes) {
+    listOfIngredients = [];
+    cartRecipes.forEach(function(recipe) {
+      // iterate through recipes in cart
+      recipe.ingredientLines.forEach(function(ingredientLine) {
+        let ingrWithoutStars = ingredientLine.replace("* ","");
+        let ingrWordsArr = ingrWithoutStars.split(" ");
+        // console.log(ingrWordsArr);
+
+        if ( ingrWordsArr.length == 1 ) {
+          // this catches instances like "salt"
+          let qty = 1;
+          let unit = 'x';
+          let desc = ingrWordsArr.slice(0, ingrWordsArr.length).join(' ');
+          listOfIngredients.push([qty, unit, desc]);
+
+        } else if ( ingrWordsArr.length == 2 && isNumber(ingrWordsArr[0]) ) {
+          // this catches instances like "1 Lime"
+          let qty = ingrWordsArr[0];
+          let unit = ''
+          let desc = ingrWordsArr.slice(1, ingrWordsArr.length).join(' ');
+          listOfIngredients.push([qty, unit, desc]);
+
+        } else {
+          // almost every other ingredient will land in here
+          let qty = ingrWordsArr[0];
```

Comment:
what happens if the ingredient is more than one word? ie. sliced apples

Reply:
We might need to do something a bit more like this psuedoCode ingrWordsArr[1:]

Reply:
good catch @lesliekimm -- this is an issue with the code as written. It's definitely not perfect right now unfortunately.

Better solutions we could try:
- check if the first word isNumber() -- > i tried this but the library i tried using wasn't always giving me true for some fractions
- find the unit of measurement (tsp, tbsp, etc.) and then assume everything before it is `qty` and everything after it is the `desc`

Reply:
something you can leave as a TODO and create another ticket for figuring out how to parse ingredients properly. that could be a ticket in itself. you will need to deal with a lot of edge cases!

Reply:
Done

Reply:
can you link the ticket here for reference?